crates/orchestrator/src/tests/jobs/mod.rs

```diff
@@ -12,11 +17,120 @@ pub mod state_update_job;
 #[rstest]
 #[tokio::test]
 async fn create_job_fails_job_already_exists() {
-    // TODO
+    todo!()
 }
 
 #[rstest]
 #[tokio::test]
 async fn create_job_fails_works_new_job() {
-    // TODO
+    todo!()
+}
+
+impl FromStr for JobStatus {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Created" => Ok(JobStatus::Created),
+            "LockedForProcessing" => Ok(JobStatus::LockedForProcessing),
+            "PendingVerification" => Ok(JobStatus::PendingVerification),
+            "Completed" => Ok(JobStatus::Completed),
+            "VerificationTimeout" => Ok(JobStatus::VerificationTimeout),
+            "VerificationFailed" => Ok(JobStatus::VerificationFailed),
+            "Failed" => Ok(JobStatus::Failed),
+            _ => Err(format!("Invalid job status: {}", s)),
+        }
+    }
+}
+
+impl FromStr for JobType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "SnosRun" => Ok(JobType::SnosRun),
+            "DataSubmission" => Ok(JobType::DataSubmission),
+            "ProofCreation" => Ok(JobType::ProofCreation),
+            "ProofRegistration" => Ok(JobType::ProofRegistration),
+            "StateTransition" => Ok(JobType::StateTransition),
+            _ => Err(format!("Invalid job type: {}", s)),
+        }
+    }
+}
+
+#[rstest]
+#[case("SnosRun", "PendingVerification")]
+#[case("DataSubmission", "Failed")]
+#[tokio::test]
+async fn handle_job_failure_job_status_typical_works(#[case] job_type: JobType, #[case] job_status: JobStatus) {
+    TestConfigBuilder::new().build().await;
+    let internal_id = 1;
+
+    let config = config().await;
+    let database_client = config.database();
+
+    // create a job
+    let mut job = build_job_item(job_type.clone(), job_status.clone(), internal_id);
+    let job_id = job.id;
+
+    // if test case is for Failure, add last_job_status to job's metadata
+    if job_status == JobStatus::Failed {
+        let mut metadata = job.metadata.clone();
+        metadata.insert("last_job_status".to_string(), "VerificationTimeout".to_string());
+        job.metadata = metadata;
+    }
+
+    // feeding the job to DB
+    database_client.create_job(job.clone()).await.unwrap();
+
+    // calling handle_job_failure
+    let response = handle_job_failure(job_id).await;
+
+    match response {
```

Comment:
let's use `assert_matches!`? i think we can match only some fields using something like this (taken from claude)

```rust
let person = Person {
        name: String::from("Alice"),
        age: 30,
        city: String::from("New York"),
    };

    assert_matches!(person, Person { name: ref n, age: 30, .. } => {
        assert_eq!(n, "Alice");
    });
```